snuba/web/views.py

```diff
@@ -373,6 +392,17 @@ def dataset_query(dataset: Dataset, body, timer: Timer, language: Language) -> R
         else:
             raise  # exception should have been chained
 
+        if language == Language.SNQL:
+            metrics.increment(
+                "snql.query.failed",
+                tags={
+                    "referrer": referrer,
+                    "type": query_type,
+                    "status": f"{status}",
+                    "message": details["message"],
```

Comment:
Please do not add a free form message as a metric tag. Since this message can include anything including timestamps, query ids, or other high cardinality data depending on the error message itself, this would generate very high cardinality tags which are expensive.